cmd/infracost/comment_bitbucket.go

```diff
@@ -0,0 +1,133 @@
+
+			var err error
+
+			serverURL, _ := cmd.Flags().GetString("bitbucket-server-url")
+			token, _ := cmd.Flags().GetString("bitbucket-token")
+			tag, _ := cmd.Flags().GetString("tag")
+			extra := comment.BitbucketExtra{
+				ServerURL: serverURL,
+				Token:     token,
+				Tag:       tag,
+			}
+
+			commit, _ := cmd.Flags().GetString("commit")
+			prNumber, _ := cmd.Flags().GetInt("pull-request")
+			repo, _ := cmd.Flags().GetString("repo")
+
+			var commentHandler *comment.CommentHandler
+			if prNumber != 0 {
+				ctx.SetContextValue("targetType", "pull-request")
+
+				commentHandler, err = comment.NewBitbucketPRHandler(ctx.Context(), repo, strconv.Itoa(prNumber), extra)
+				if err != nil {
+					return err
+				}
+			} else if commit != "" {
+				ctx.SetContextValue("targetType", "commit")
+
+				commentHandler, err = comment.NewBitbucketCommitHandler(ctx.Context(), repo, commit, extra)
+				if err != nil {
+					return err
+				}
+			} else {
+				ui.PrintUsage(cmd)
+				return fmt.Errorf("either --commit or --pull-request is required")
+			}
+
+			behavior, _ := cmd.Flags().GetString("behavior")
+			if behavior != "" && !contains(validCommentBitbucketBehaviors, behavior) {
+				ui.PrintUsage(cmd)
+				return fmt.Errorf("--behavior only supports %s", strings.Join(validCommentBitbucketBehaviors, ", "))
+			}
+			ctx.SetContextValue("behavior", behavior)
+
+			paths, _ := cmd.Flags().GetStringArray("path")
+
+			body, err := buildCommentBody(cmd, ctx, paths, output.MarkdownOptions{
+				WillUpdate:          prNumber != 0 && behavior == "update",
+				WillReplace:         prNumber != 0 && behavior == "delete-and-new",
+				IncludeFeedbackLink: true,
+				BasicSyntax:         true,
+			})
+			if err != nil {
+				return err
+			}
+
+			dryRun, _ := cmd.Flags().GetBool("dry-run")
+			if !dryRun {
+				err = commentHandler.CommentWithBehavior(ctx.Context(), behavior, string(body))
+				if err != nil {
+					return err
+				}
+
+				pricingClient := apiclient.NewPricingAPIClient(ctx)
+				err = pricingClient.AddEvent("infracost-comment", ctx.EventEnv())
+				if err != nil {
+					log.Errorf("Error reporting event: %s", err)
+				}
+
+				cmd.Println("Comment posted to Bitbucket")
+			} else {
+				cmd.Println(string(body))
+				cmd.Println("Comment not posted to Bitbucket (--dry-run was specified)")
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String("behavior", "update", `Behavior when posting comment, one of:
+  update (default)  Update latest comment
+  new               Create a new comment
+  delete-and-new    Delete previous matching comments and create a new comment`)
+	_ = cmd.RegisterFlagCompletionFunc("behavior", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return validCommentBitbucketBehaviors, cobra.ShellCompDirectiveDefault
+	})
+	cmd.Flags().String("bitbucket-server-url", "", "Bitbucket Server URL")
+	cmd.Flags().String("bitbucket-token", "", "Bitbucket access token")
```

Comment:
```suggestion
	cmd.Flags().String("bitbucket-token", "", "Bitbucket access token. Use 'user:password' for Bitbucket Cloud and HTTP access token for Bitbucket Server.")
```

Is this right? We could also link to https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/ in the docs repo and the Bitbucket readme

Reply:
Yep, `password` is the App password, maybe we can show it as `user:app-password`.

Reply:
good idea, I updated my original suggestion